akka-actor/src/main/scala/akka/pattern/HandleBackoff.scala

```diff
@@ -47,8 +48,8 @@ import akka.annotation.InternalApi
         case msg         => unhandled(msg)
       }
 
-    case ResetRestartCount(current) =>
-      if (current == restartCount) {
+    case ResetRestartCount(seenRestartCount) =>
+      if (seenRestartCount == restartCount && restartCount > 0) {
```

Comment:
maybe it would be simplified by using Timers api, but maybe we don't want to touch this too much?

Reply:
Yeah, let's not open a can of worms.